log/enabled.go

```diff
@@ -0,0 +1,19 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package log // import "go.opentelemetry.io/otel/log"
+
+// EnabledOpts represents payload for [Logger]'s Enabled method.
+type EnabledOpts struct {
```

Comment:
This does not follow our options pattern policy: https://github.com/open-telemetry/opentelemetry-go/blob/main/CONTRIBUTING.md#option

Please update to match or use a different name here to distinguish the alternate intent.

Reply:
Which part of the policy does it violate?  Is your main concern user confusion?  Or that it would introduce a new "pattern" which others would want to adopt?

Another idea would be to choose something smilar to Record (since this is details about the log record): RecordDetails, RecordSummary, RecordXXX?  Or we could go back to params

Reply:
We have a definition for what an "option" is. We need to follow closer to what we do for [`SamplingParameters`](https://pkg.go.dev/go.opentelemetry.io/otel/sdk/trace#SamplingParameters), including naming.

Reply:
Should I revert to EnabledParam or change to EnabledParameters?

Reply:
+1 to `EnabledParameters` to match the tracing signal. But, I'm not opposed to `EnabledParam`.

Reply:
Given this isn't a 1-to-1 matching with the trace SDK's `Sampler` interfaced I think we have some leeway in naming. Just, we need to not conflict with our options naming or pattern.

Reply:
I prefer shorter `EnabledParam`, but I am also not opposed to `EnabledParameters`.

I propose a vote for probing.

- 🚀 for `EnabledParam`
- 🎉 for `EnabledParameters`

Reply:
Renamed to `EnabledParameters`. b9a617178